### PR TITLE
Application updates

### PR DESCRIPTION
**Pull request by r-scheele**
- Fix git operation class to handle a push to gitlab in docker container
- Fix tests to account for updates in the route
- Added postgres service, Ensure pytest and black formatter works, changed package manager to poetry in the workflow

### CODE REVIEW

File: app/server/routes/policies.py [hunk 1]
```diff
@@ -34,7 +34,9 @@ async def write_policy(
     policies.append(policy)
 
     # Write the policy to the database after successful push
-    WriteRego(dependencies["token"], rego_rule.repo_url).write_to_file(policies)
+    WriteRego(
+        dependencies["token"], policy["repo_url"], dependencies["login"]
```
Comment (Youngestdev) [inline app/server/routes/policies.py:38]: format to be on the same line.

File: app/server/routes/policies.py [hunk 2]
```diff
@@ -73,9 +75,10 @@ async def modify_policy(
 
     policies = database.get_policies(dependencies["login"])
     policies.append(updated_policy)
-
     # Rewrite rego file and update GitHub
-    WriteRego(dependencies["token"], rego_rule["repo_url"]).write_to_file(policies)
+    WriteRego(
```
Comment (Youngestdev) [inline app/server/routes/policies.py:79]: format to be on a single line.

File: app/server/routes/policies.py (hunk 1 above)
Comment (r-scheele) [reply app/server/routes/policies.py:38]: Black formatted it that way🤔

File: app/server/routes/policies.py (hunk 2 above)
Comment (r-scheele) [reply app/server/routes/policies.py:79]: Black formatted it that way🤔

Comment (Youngestdev) [reply app/server/routes/policies.py:79]: oh ok

**Pull request merged**